src/plugins/discover/public/application/main/components/sidebar/discover_sidebar_responsive.tsx

```diff
@@ -111,38 +114,54 @@ export interface DiscoverSidebarResponsiveProps {
  */
 export function DiscoverSidebarResponsive(props: DiscoverSidebarResponsiveProps) {
   const services = useDiscoverServices();
+  const { data, dataViews, core } = services;
   const isPlainRecord = useAppStateSelector(
     (state) => getRawRecordType(state.query) === RecordRawType.PLAIN
   );
   const { selectedDataView, onFieldEdited, onDataViewCreated } = props;
   const [fieldFilter, setFieldFilter] = useState(getDefaultFieldFilter());
   const [isFlyoutVisible, setIsFlyoutVisible] = useState(false);
-  /**
-   * fieldCounts are used to determine which fields are actually used in the given set of documents
-   */
-  const fieldCounts = useRef<Record<string, number> | null>(null);
-  if (fieldCounts.current === null) {
-    fieldCounts.current = calcFieldCounts(props.documents$.getValue().result!, selectedDataView);
-  }
+  const [allFields, setAllFields] = useState<DataViewField[] | null>(null);
+  const [fieldCounts, setFieldCounts] = useState<Record<string, number> | null>(null);
+  const dataViewFields = selectedDataView?.fields;
 
-  const [documentState, setDocumentState] = useState(props.documents$.getValue());
   useEffect(() => {
-    const subscription = props.documents$.subscribe((next) => {
-      if (next.fetchStatus !== documentState.fetchStatus) {
-        if (next.result) {
-          fieldCounts.current = calcFieldCounts(next.result, selectedDataView!);
-        }
-        setDocumentState({ ...documentState, ...next });
+    const subscription = props.documents$.subscribe((documentState) => {
+      if (documentState?.fetchStatus === FetchStatus.COMPLETE) {
+        setFieldCounts(calcFieldCounts(documentState.result));
+      } else {
+        setFieldCounts(null);
       }
     });
     return () => subscription.unsubscribe();
-  }, [props.documents$, selectedDataView, documentState, setDocumentState]);
+  }, [props.documents$, setAllFields, setFieldCounts]);
```

Comment:
what if we handle the field existence more imperative, when documents$ emits FetchStatus.LOADING, refetchFieldsExistenceInfo() is executed, if necessary. Also in this file could be the place where we get / store the time range using the timefilter function

```
export const getResolvedDateRange = function (timefilter: TimefilterContract) {
  const { from, to } = timefilter.getTime();
  const { min, max } = timefilter.calculateBounds({
    from,
    to,
  });
  return { fromDate: min?.toISOString() || from, toDate: max?.toISOString() || to };
};
```
We wouldn't fetch the fields initially, this is also not the case currently, where the fields are linked to the returned documents. The sections currently would display the loading spinner, those should only be displayed when the fields are loaded, Lens loads initially, in Discover this is configurable